src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -687,9 +678,7 @@ def tool_message_parts(
             )
 
     message: str
-    if isinstance(
-        requirements, (ToolCustomLockfile, ToolDefaultLockfile, ToolProgrammaticLockfile)
-    ):
+    if isinstance(requirements, (ToolCustomLockfile, ToolDefaultLockfile)):
         message = "".join(tool_message_parts(requirements)).strip()
     else:
         message = "TODO: add support for user lockfiles"
```

Comment:
This can be removed. The funciton only supports tool lockfiles, meaning that you can't hook up the invalidation check for anything else in Pants 2.7.

Reply:
I've replaced with a comment and a false assertion